monitoring/bin/deploy_monitoring_tenant.sh

```diff
@@ -0,0 +1,170 @@
+for f in $tenantDir/*; do
+  if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
+    sed -i '' "s/__TENANT__/$VIYA_TENANT/g" $f
+    sed -i '' "s/__TENANT_NS__/$VIYA_NS/g" $f
+    sed -i '' "s/__MON_NS__/$MON_NS/g" $f
+  else
+    sed -i "s/__TENANT__/$VIYA_TENANT/g" $f
+    sed -i "s/__TENANT_NS__/$VIYA_NS/g" $f
+    sed -i "s/__MON_NS__/$MON_NS/g" $f
+  fi
+done
+
+# Grafana options
+grafanaYAML=$tenantDir/mt-grafana-values.yaml
+
+# TLS
+grafanaTLSYAML=$TMP_DIR/empty.yaml
+if [ "$MON_TLS_ENABLE" == "true" ] || [ "$TLS_ENABLE" == "true" ]; then
+  grafanaTLSYAML=$tenantDir/mt-grafana-tls-values.yaml
+fi
+
+# USER_DIR customizations
+userGrafanaYAML=$TMP_DIR/empty.yaml
+if [ -f "$USER_DIR/monitoring/user-values-multitenant-grafana.yaml" ]; then
+  userGrafanaYAML="$USER_DIR/monitoring/user-values-multitenant-grafana.yaml"
+  log_debug "User response file for Grafana found at [$userGrafanaYAML]"
+fi
+
+# Optional workload node placement support
+MON_NODE_PLACEMENT_ENABLE=${MON_NODE_PLACEMENT_ENABLE:-${NODE_PLACEMENT_ENABLE:-false}}
+if [ "$MON_NODE_PLACEMENT_ENABLE" == "true" ]; then
+  log_info "Enabling monitoring components for workload node placement"
+  wnpGrafanaValuesFile="monitoring/node-placement/values-grafana-wnp.yaml"
+else
+  log_debug "Workload node placement support is disabled"
+  wnpGrafanaValuesFile="$TMP_DIR/empty.yaml"
+fi
+
+# Deploy additional scrape configs for Prometheus
+log_info "Creating scrape config secret"
+kubectl delete secret --ignore-not-found -n $VIYA_NS prometheus-federate-$VIYA_TENANT
+kubectl create secret generic \
+  -n $VIYA_NS \
+  prometheus-federate-$VIYA_TENANT \
+  --from-file cluster-federate-job=$tenantDir/mt-federate-secret.yaml
+
+# Deploy Prometheus
+log_info "Deploying Prometheus"
+kubectl apply -n $VIYA_NS -f $tenantDir/mt-prometheus.yaml
+
+# Grafana password
+if helm3ReleaseExists grafana-$VIYA_TENANT $VIYA_NS; then
+  log_info "Upgrading Grafana via Helm...($(date) - timeout 10m)"
+else
+  grafanaPwd="$GRAFANA_ADMIN_PASSWORD"
+  if [ "$grafanaPwd" == "" ]; then
+    log_debug "Generating random Grafana admin password..."
+    showPass="true"
+    grafanaPwd="$(randomPassword)"
+  fi
+  log_info "Deploying Grafana via Helm...($(date) - timeout 10m)"
+fi
+
+# Deploy Grafana
+GRAFANA_CHART_VERSION_TENANT=${GRAFANA_CHART_VERSION_TENANT:-6.9.1}
+helm upgrade --install $helmDebug \
+  -n "$VIYA_NS" \
+  -f "$wnpGrafanaValuesFile" \
+  -f "$grafanaYAML" \
+  -f "$grafanaTLSYAML" \
+  -f "$userGrafanaYAML" \
+  --set "extraLabels.sas\\.com/tenant=$VIYA_TENANT" \
+  --set grafana.adminPassword="$grafanaPwd" \
+  --version "$GRAFANA_CHART_VERSION_TENANT" \
+  --atomic \
+  --timeout "10m" \
+  $extraArgs \
+  grafana-$VIYA_TENANT \
+  grafana/grafana
+
+# Deploy ServiceMonitors
+kubectl apply -n $VIYA_NS -f $tenantDir/serviceMonitor-sas-cas-tenant.yaml
+kubectl apply -n $VIYA_NS -f $tenantDir/serviceMonitor-sas-pushgateway-tenant.yaml
+
+function deploy_tenant_dashboards {
+   dir=$1
```

Comment:
Possible future refactoring opportunity: call the (existing) monitoring/bin/deploy_dashboards.sh script after supporting any tenant-specific needs.